### PR TITLE
Adding DEB Maintainers Scripts and Build CI Images

### PR DESCRIPTION
**Pull request by peterzhuamazon**
### Description
Adding DEB Maintainers Scripts and Build CI Images

### Issues Resolved
https://github.com/opensearch-project/opensearch-build/issues/28

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by zelinh — COMMENTED** [review]
`jenkins/opensearch/.deb.jenkinsfile.swp` Is this file committed accidentally?

**Comment by peterzhuamazon** [comment]
> `jenkins/opensearch/.deb.jenkinsfile.swp` Is this file committed accidentally?

Thanks, removed now.a

File: jenkins/opensearch/distribution-build.jenkinsfile [hunk 1]
```diff
@@ -531,6 +532,86 @@ pipeline {
...
                     }
                 }
+                stage('build-and-test-linux-x64-deb') {
+                    when {
+                        beforeAgent true
+                        expression{
+                            params.BUILD_PLATFORM.contains('linux')
+                        }
+                    }
+                    agent { label AGENT_X64 }
```
Comment (gaiksaya) [inline jenkins/opensearch/distribution-build.jenkinsfile:542]: I believe the agent would be Ubuntu right? Or does AL2 work here too?

File: scripts/pkg/build_templates/opensearch-dashboards/deb/debian/control [hunk 2]
```diff
@@ -8,5 +15,6 @@ Homepage: https://opensearch.org/
 
 Package: opensearch-dashboards
 Architecture: any
-Multi-Arch: foreign
-Description: OpenSearch
+Description: Open source visualization dashboards for OpenSearch
+    OpenSearch Dashboards is the visualization tool for data in OpenSearch
```
Comment (gaiksaya) [inline scripts/pkg/build_templates/opensearch-dashboards/deb/debian/control:19]: Are both there lines required? They almost mean the same.

**Review by gaiksaya — COMMENTED** [review]
Hi @peterzhuamazon ,
Can you please rebase to 1-2 commits? Looks like this PR has 115 commits.
Thanks! Added minor comments.

File: jenkins/opensearch/distribution-build.jenkinsfile (hunk 1 above)
Comment (peterzhuamazon) [reply jenkins/opensearch/distribution-build.jenkinsfile:542]: agent is AL2, docker is ubuntu2004 😄

File: scripts/pkg/build_templates/opensearch-dashboards/deb/debian/control (hunk 2 above)
Comment (peterzhuamazon) [reply scripts/pkg/build_templates/opensearch-dashboards/deb/debian/control:19]: Required, the 1st line is effectively the same as RPM summary, and will disable only the short line when using dpkg -l.
So yes both are required as that is how debian design their packages.

File: jenkins/opensearch/distribution-build.jenkinsfile (hunk 1 above)
Comment (peterzhuamazon) [reply jenkins/opensearch/distribution-build.jenkinsfile:542]: Also be aware build is happening on centos7 docker, assemble is happening on ubuntu2004 docker.

Comment (gaiksaya) [reply jenkins/opensearch/distribution-build.jenkinsfile:542]: Is that intended? Or should build be happening on ubuntu too?

Comment (peterzhuamazon) [reply jenkins/opensearch/distribution-build.jenkinsfile:542]: Build should definitely not happening on Ubuntu, it should happen on centos7 as it has the lowest version of glibc.
Then rpm will assemble on rockylinux8 and deb will assemble on ubuntu2004.
This is by design and it is how it should run.

Comment (gaiksaya) [reply jenkins/opensearch/distribution-build.jenkinsfile:542]: Thanks!

**Review by rishabh6788 — APPROVED** [review]

**Comment by peterzhuamazon** [comment]
> > `jenkins/opensearch/.deb.jenkinsfile.swp` Is this file committed accidentally?
> 
> Thanks, removed now.

Since there are a lot of commits and cause a lot of conflicts during the merge, we decided to keep it as it is.
Tho next time we would make sure it is rebased before pushing to PR.

Thanks.

**Pull request merged**